test/code/plugins/VMInsightsDataCollector_subtest.rb

```diff
@@ -1127,27 +1130,29 @@ def initialize(dev, mp, size, free, type=(Random.rand(3) + 2))
             end
 
             def to_s
-                "%-10s %-15s %15d %15d" % [ @dev, @mp, @size, @free ]
+                "%-10s %-15s %15d %15d %-10s" % [ @dev, @mp, @size, @free, @type ]
             end
 
             def equivilent?(actual)
                 @dev == '/dev/' + actual.device_name &&
                 @mp == actual.mount_point &&
                 @size == actual.size_in_bytes &&
-                @free == actual.free_space_in_bytes
+                @free == actual.free_space_in_bytes &&
+                @type == actual.filesystem
             end
 
             def <=>(o)
```

Comment:
Given the existing bug noted below and the new logic error, I have to wonder if this routine is actually used.